Sources/Hummingbird/Codable/JSON/JSONCoding.swift

```diff
@@ -23,12 +23,15 @@ extension JSONEncoder: ResponseEncoder {
     ///   - value: Value to encode
     ///   - request: Request used to generate response
     public func encode(_ value: some Encodable, from request: Request, context: some BaseRequestContext) throws -> Response {
-        var buffer = context.allocator.buffer(capacity: 0)
         let data = try self.encode(value)
+        var buffer = context.allocator.buffer(capacity: data.count)
```

Comment:
Can we use NIO's allocator helpers here

Reply:
[For bytes](https://github.com/apple/swift-nio/blob/609469f9689603c8419f1402dd984b0faa763c54/Sources/NIOCore/ByteBuffer-aux.swift#L739-L742)

Reply:
Fair point. Will fix